engine/scripts.py

```diff
@@ -46,6 +48,9 @@ def create_sprite(
     ) -> None:
         """Creates a sprite."""
 
+    def get_sprites(self, name: str) -> Iterable[Sprite]:
+        """Gets all sprites with the given name."""
```

Comment:
I have made the change to use `Optional[str]`. However, I don't understand how it will be used to get the sprites by location. Does location is an area (of the given shape) that, say a monster, looks in for the player?

Reply:
Yeah exactly. We're not implementing that yet, we'll do it later.